### PR TITLE
[skflow] Added support for different parameters per layer in dnn

### PR DESCRIPTION
**Pull request by nmiculinic**
Added support for per layer configuration of activation function and
dropout probabilities. Example usage as per following code:

``` python
def model(X, y):
    l3 = skflow.ops.dnn(
        X,
        hidden_units=[10, 20],
        activation=[tf.nn.tanh, tf.nn.relu6]
    )
    return skflow.models.linear_regression(l3, y)

regressor = skflow.TensorFlowEstimator(
    model_fn=model,
    n_classes=0,
    steps=50000,
    learning_rate=0.1,
    batch_size=128,
    verbose=1
)

regressor.fit(X_t, Y_t)
```

Tested with single activation function and with multiple ones that
nothing crashes. Seem to be working correctly

### CODE REVIEW

**Comment by vrv** [comment]
ping for @ilblackdragon  (i hope you are getting these mentions)

File: tensorflow/contrib/learn/python/learn/ops/dnn_ops.py [hunk 1]
```diff
@@ -28,14 +28,43 @@ def dnn(tensor_in, hidden_units, activation=nn.relu, dropout=None):
...
+    if len(f) != length:
+      raise ValueError("invalid supplied argument size")
+    return f
+
+  activation = broadcast_or_noop(activation, len(hidden_units))
+  dropout = broadcast_or_noop(dropout, len(hidden_units), None)
+
   with vs.variable_scope('dnn'):
-    for i, n_units in enumerate(hidden_units):
+    for i, (n_units, activation_l, dropout_l) in \
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/ops/dnn_ops.py]: may be `activation_fn` and `dropout_prob` ?

**Comment by ilblackdragon** [comment]
Ok, we plan to move / change a bit this function, so it would be good to have this PR in for informing use cases.

**Comment by ilblackdragon** [comment]
@tensorflow-jenkins test this please!

**Comment by rmlarsen** [comment]
Ping for @ilblackdragon. Is this still in progress?

**Comment by ilblackdragon** [comment]
@rmlarsen when @nmiculinic addressed one naming comment it can go in.

**Comment by vrv** [comment]
Closing due to inactivity.  Ping if ready to re-open / rebase to master.
